### PR TITLE
Refactor Engine State 

### PR DESCRIPTION
**Pull request by passabilities**
### Merge Code Paths
The engine state has been refactored which simplifies the flow of into smaller modular functions. By doing this, the code paths that are executed when processing and address for the first time and while an address is subscribed to for new transactions is the same.

### CODE REVIEW

File: src/common/utxobased/engine/makeUtxoEngineState.ts [hunk 1]
```diff
@@ -431,3 +437,89 @@ const processAddressBalance = async (args: ProcessAddressBalanceArgs): Promise<v
...
+  const scriptPubkey = walletTools.addressToScriptPubkey(address)
+  const addressData = await processor.fetchAddressByScriptPubkey(scriptPubkey)
+  let networkQueryVal = args.networkQueryVal ?? addressData?.networkQueryVal
+  const {
+    transactions = [],
+    totalPages
+  } = await blockBook.fetchAddress(address, {
+    details: 'txs',
+    from: networkQueryVal,
+    perPage: 10,
```
Comment (passabilities) [inline src/common/utxobased/engine/makeUtxoEngineState.ts:472]: set variable in constants file

File: src/common/utxobased/engine/makeUtxoEngineState.ts [hunk 2]
```diff
@@ -394,7 +395,8 @@ const processAddress = async (args: ProcessAddressArgs) => {
 
   await Promise.all([
     processAddressBalance(args),
```
Comment (passabilities) [inline src/common/utxobased/engine/makeUtxoEngineState.ts:397]: process balance in the utxo function

**Review by thehobbit85 — CHANGES_REQUESTED** [review]
Requested changes

File: src/common/utxobased/engine/makeUtxoEngineState.ts [hunk 3]
```diff
@@ -46,332 +56,469 @@ export function makeUtxoEngineState(config: UtxoEngineStateConfig): UtxoEngineSt
...
-      const rawTx = await blockBook.fetchTransaction(txid)
-      tx = processRawTransaction(rawTx)
-    }
-    return tx
+  for (const rawTx of transactions) {
+    const tx = processRawTx({ ...args, tx: rawTx })
+    processor.saveTransaction(tx)
   }
 
-  async function onNewBlock(): Promise<void> {
```
Comment (TheCharlatan) [inline src/common/utxobased/engine/makeUtxoEngineState.ts:274]: How is this handled now?

**Review by thehobbit85 — APPROVED** [review]

**Pull request merged**